extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandlerBase.java

```diff
@@ -56,13 +70,34 @@ private void doHandle(RoutingContext ctx, ManagedContext requestContext) {
                 requestContext.terminate();
             }
             HttpServerResponse resp = ctx.response();
+            Buffer buffer = Buffer.buffer(256); // this size seems to cover the basic health checks
             if (health.isDown()) {
                 resp.setStatusCode(503);
+                if (problemDetails) {
+                    resp.headers().set(HttpHeaders.CONTENT_TYPE, "application/problem+json");
+                    try (BufferOutputStream outputStream = new BufferOutputStream(buffer)) {
+                        JsonObjectBuilder objectBuilder = JSON_PROVIDER.createObjectBuilder();
+                        objectBuilder
+                                .add("type", "about:blank")
+                                .add("status", 503)
+                                .add("title", "Health Check Failed: " + ctx.normalizedPath())
+                                .add("detail", ctx.request().uri() + ", invoked at " + Instant.now())
+                                .add("instance", ctx.request().absoluteURI())
+                                .add("health", health.getPayload());
+                        JsonWriterFactory factory = JSON_PROVIDER.createWriterFactory(JSON_CONFIG);
```

Comment:
Can't this factory be created only once for the class?

Reply:
It is created only here. Not sure what you mean.